### DISCUSSION TITLE
Rete nattata e NGROK

### DISCUSSION THREAD

**Discussion by Zatarraz** [indomus/forum#20] [Networking]
Ciao a tutti.
Una domanda: sono riuscito a impostare NGROK per accedere ad HA da remoto sulla mia rete nattata, ma ieri per cause di forza maggiore ho dovuto togliere corrente a tutta la casa e come d'incanto NGROK mi ha cambiato il tunnel e, di conseguenza, ho perso l'accesso ad HA da remoto.
Rileggendo bene sul sito di NGR mi sono reso conto che per avere accesso sempre dallo stesso tunnel dovrei pagare 5$ al mese.
A questo punto credo che sia meglio pagare il mio fornitore di servizi internet ed avere IP pubblico per accedere da remoto a HA e magari anche ad altro.
Sembra che me la sono suonata e cantata da solo, ma tutto ciò solo per chiedervi se è corretto ciò che ho dedotto, oppure non ci ho capito una fava di NGROK.
Grazie per le risposte.

**Comment by snakuzzo** [comment]
Non ho mai usato NGROK, ma se proprio devi pagare...sono d'accordo con te. A quel punto paghi per avere un IP pubblico, non per avere un servizio per aggirare il fatto che tu sia nattato.
Altra cosa...hai provato a dare un'occhiata a remote.it ?

**Reply by Zatarraz** [reply]
Grazie per la prontissima risposta.
Prima di investire 5€/mese sicuramente le proverò tutte.
Non è per i 5€, ma per principio: se esiste il modo, perchè pagare?

**Reply by snakuzzo** [reply]
ma sì...ti comprendo benissimo.
Fai benissimo a provare soluzioni alternative ;)

**Reply by Zatarraz** [reply]
Sto provando a capirci qualcosa, ma non riesco nemmeno a capire come installarlo su home assistant...

**Reply by snakuzzo** [reply]
perché non è un addon di HA...se hai HASSIO non so cosa consigliarti sinceramente...per me HASSIO non esiste...esiste solo HA Core :)

**Reply by Zatarraz** [reply]
Io ho installato Home Assistant os, che probabilmente è HASSIO, giusto?

**Reply by xyzmauro** [reply]
Comunque non hai perso l’accesso da remoto,solo ti ha cambiato il link da cui collegarti e nel profilo gratuito lo fa ogni volta che chiudi il tunnel,per esempio con un riavvio del Raspberry...io lo utilizzavo tramite Node-Red e aprivo il tunnel quando mi serviva e lo chiudevo quando avevo finito tramite command da Telegram e in risposta avevo il link da utilizzare,se ti interessa cerco il flow e te lo posto

**Reply by xyzmauro** [reply]
Come non detto, letto dopo che usi HASSIO e quindi avresti Node-red come add-on e non credo funzioni quel nodo di Ngrok perchè apre il tunnel su localhost (almeno, io con Core su docker non son riuscito a farlo funzionare)

**Reply by Zatarraz** [reply]
Buonasera a tutti.
Dopo aver usato NGROK per un po' di tempo, ho deciso di aprire ip pubblico.
A questo punto, ho pensato di accedere ad HASSIO da remoto con Duckdns e non più con NGROK, in quanto ogni volta che per qualche motivo il raspberry si disconnette, devo reimpostare l'indirizzo e, fino a quel momento, non ho accesso dall'esterno. 
Impostato Duckdns, ho provato a collegarmi da remoto ad Hassio e qui sorge un nuovo problema: se digito l'indirizzo di HA, invece di entrare nel login di HA, mi fa entrare nel login del modem di Linkem.
Avendo io una casa su tre livelli, dopo il modem di linkem ho collegato a cascata un router Fritzbox, in modo tale da avere la possibilità di una rete MESH.
Il modem di linkem ha come indirizzo di rete 192.168.1.xxx, mentre il Fritz e tutta la rete hanno 192.168.178.xxx.
Naturalmente il modem di linkem mi può aprire le porte solo della rete 192.168.1.xxx.
Spero di essere stato chiaro nella descrizione del problema.
Qualcuno ha qualche idea per potermi aiutare?

**Reply by snakuzzo** [reply]
Premesso che il mesh di AVM fa un po' ca....are, come stai facendo tu hai un doppio NAT...e questa cosa non va bene.
Dovresti provare a vedere se il modem linkem prevede una modalità bridged e non routed, in modo tale da usarlo solo come modem e non come router e configurare di conseguenza il Frtiz per avviare una connessione PPPoE.

In alternativa, dovresti aprire le porte due volte.
Es.

Apri porta ext 8123 Linkem -> 8123 IP WAN del Fritzbox    e poi   8123 Fritzbox -> 8123 ed IP LAN del raspberry

**Reply by Scialla** [reply]
Io uso due router (quello del modem e quello della Time Capsule) sullo stesso range di IP e bypasso questo problema... lo so che non si dovrebbe fare ma funziona. Il router del modem da 0 a 50, quello della TC da 51 a 247.
Ho dovuto fare così perché do connessione al vicino a cui serve il DHCP del modem.
Però mi sembra di ricordare di avere fatto prove col Fritz per fare lo stesso ed avere riscontrato problemi, boh

**Comment by Zatarraz** [comment]
Ciao.
Intanto grazie mille per la risposta. Ho provato ad aprire la doppia porta,
ma mi ferma subito: il modem di Linkem è abbastanza chiuso e mi fa aprire
solo porte verso indirizzi 192.168.1.XXX, mentre l'indirizzo ip del router
AVM è 192.168.178.xxx. Non so come venirne fuori fino a quando non cambierò
operatore...

Il gio 27 mag 2021, 11:12 Diego Ambrosanio ***@***.***> ha
scritto:
